apps/costcontrol/js/message_handler.js

```diff
@@ -323,26 +299,47 @@
     });
   }
 
+  function sendNotification(activity, title, message, notificationData) {
+    // XXX: Hack hiding the message class in the icon URL
+    // Should use the tag element of the notification once the final spec
+    // lands:
+    // See: https://bugzilla.mozilla.org/show_bug.cgi?id=782211
+    navigator.mozApps.getSelf().onsuccess = function _onAppReady(evt) {
+      var app = evt.target.result;
+      var iconURL = NotificationHelper.getIconURI(app);
+      if (!notificationData) {
+        notificationData = {'type': 'dataUsage'};
+      }
+      var goTo;
+      if (!inStandAloneMode()) {
+        goTo = function _goTo() {
+          if (inApplicationMode()) {
+            app.launch();
+            activity.action && activity.action();
+          } else {
+            activity = new MozActivity({name: activity.name});
```

Comment:
I would preffer to rename `activity` to `action`. This way `action` could refer to an **activity** or a **callback**, depending on which scenario are we.